log.go

```diff
@@ -76,33 +77,60 @@ type LeveledLogger struct {
 	stdoutOverride io.Writer
 }
 
+var apiKeyRegex *regexp.Regexp
+
+func init() {
+	var err error
+	apiKeyRegex, err = regexp.Compile("(sk_live|sk_test|rk_live|rk_test)_[a-zA-Z0-9]{2}[a-zA-Z0-9]{1,}")
+	if err != nil {
+		panic(err)
+	}
+}
+
+// Redact the middle of a string as needed and replace it with an asterisk.
+// Keep the given number of characters at the start and end of the string.
+func redactMiddle(msg string, startLen int, endLen int) string {
+	if len(msg) <= startLen+endLen {
+		return msg
+	}
+	redactedSubstr := ""
+	for i := 0; i < len(msg)-endLen-startLen; i++ {
+		redactedSubstr += "*"
+	}
+	return msg[:startLen] + redactedSubstr + msg[len(msg)-endLen:]
+}
+
+func redact(msg string) string {
+	return apiKeyRegex.ReplaceAllStringFunc(msg, func(str string) string {
+		return redactMiddle(str, 10, 4)
+	})
+}
+
+func (l *LeveledLogger) printLog(format string, tag string, level Level, output io.Writer, v ...interface{}) {
+	if l.Level >= level {
+		s := fmt.Sprintf(format, v...)
+		fmt.Fprintf(output, "[%s] %s\n", tag, redact(s))
+	}
+}
```

Comment:
I've tried to keep files like this organized so that you have the public-facing types/functions nearer to the top, and the internal ones below (see comments like "Public types". Can we move all of this down to a new section at the bottom (except `printLog` which belongs on `LeveledLogger`, but try to order it alphabetically so it appears below `Warnf` and above `stderr`.